mesonbuild/backend/ninjabackend.py

```diff
@@ -485,13 +485,14 @@ def __init__(self, build: T.Optional[build.Build], interpreter: T.Optional[Inter
         self.name = 'ninja'
         self.ninja_filename = 'build.ninja'
         self.fortran_deps = {}
-        self.all_outputs = {}
+        self.all_outputs: T.Dict[str, Literal[True]] = {}
```

Comment:
Shouldn't that be changed to `T.Set[str]` while at it?

Reply:
You mean change the code to use a set instead of a True-keyed dict?

Reply:
Yes, unrelated to this typing PR, but if we don't cleanup that now we'll never do it :P

Reply:
Should be fixed.